xip/xipether.c

```diff
@@ -17,13 +17,116 @@
 #include "libnetlink.h"
 #include "xiart.h"
 
+#define IFINDEX_STR_SIZE 8
+#define XIA_LLADDR_LEN 12
+#define APPEND_ZEROES 20
+
 static int usage(void)
 {
 	fprintf(stderr,
-"Usage:	\n");
+"Usage:	xip ether { addinterface | delinterface } dev DEV\n"
+"       xip ether { addneigh | delneigh } lladdr LLADDR dev DEV\n"
+"       xip ether show { interfaces | neighs }\n"
+"where	LLADDR := HEXDIGIT{1,2} (':' HEXDIGIT{1,2})*\n"
+"	DEV := STRING NUMBER\n");
 	return -1;
 }
 
+static void form_ether_xid(unsigned oif, unsigned char *lladdr, unsigned tlen, char *id)
+{
+	char bytes[4];
```

Comment:
Why not declare it as ``unsigned char bytes[4]``? This way, you may not need the cast below.